docs/_ext/sphinx_science/directives.py

```diff
@@ -143,6 +143,10 @@ class Synthesized(Directive):
             optional_arguments = getattr(doc_gen_directive, "optional_arguments", 0)
 
             def run(self) -> list[nodes.Node]:
+                assert self.state.document.current_source is not None, (
```

Comment:
This was a new MyPy warning. I'm not sure why this did not fire before, all the dev deps to run linting, checking, etc are locked and did not change in this PR.